backend/onyx/document_index/vespa/index.py

```diff
@@ -307,47 +314,61 @@ def register_multitenant_indices(
     def index(
         self,
         chunks: list[DocMetadataAwareIndexChunk],
-        fresh_index: bool = False,
     ) -> set[DocumentInsertionRecord]:
-        """Receive a list of chunks from a batch of documents and index the chunks into Vespa along
-        with updating the associated permissions. Assumes that a document will not be split into
-        multiple chunk batches calling this function multiple times, otherwise only the last set of
-        chunks will be kept"""
+        """
+        Index a list of chunks into Vespa, optionally removing older chunk versions.
+        Raises an AssertionError if multiple chunks for the same doc_id have conflicting doc_updated_at values.
+        """
+        current_timestamp = int(datetime.utcnow().timestamp())
+
+        # Step 1: Enforce a single doc_updated_at for each doc_id across all chunks.
+        doc_to_doc_updated_at = {}
+        for chunk in chunks:
+            doc_id = chunk.source_document.id
+
+            chunk_val = chunk.source_document.doc_updated_at
+            if doc_id not in doc_to_doc_updated_at:
+                # If we haven't encountered this doc_id yet, store or fill its doc_updated_at
+                if chunk_val is None:
+                    # If missing, default to "now"
+                    chunk_val = datetime.fromtimestamp(
+                        current_timestamp, tz=timezone.utc
+                    )
+                    chunk.source_document.doc_updated_at = chunk_val
+                doc_to_doc_updated_at[doc_id] = chunk_val
+            else:
+                # If the chunk's doc_updated_at is missing, fill it from the stored value
+                if chunk_val is None:
+                    chunk.source_document.doc_updated_at = doc_to_doc_updated_at[doc_id]
+                else:
+                    # Raise if doc_updated_at conflicts with what we've already stored for this doc_id
+                    assert chunk_val == doc_to_doc_updated_at[doc_id], (
```

Comment:
we probably mean to raise an exception here

Reply:
Good catch